linkerd/app/core/src/trace.rs

```diff
@@ -33,25 +45,52 @@ pub fn init_log_compat() -> Result<(), Error> {
     tracing_log::LogTracer::init().map_err(Error::from)
 }
 
-pub fn with_filter(filter: impl AsRef<str>) -> (Dispatch, LevelHandle) {
+pub fn with_filter_and_format(
+    filter: impl AsRef<str>,
+    format: impl AsRef<str>,
+) -> (Dispatch, LevelHandle) {
     let filter = filter.as_ref();
+    let format = format.as_ref();
 
     // Set up the subscriber
     let start_time = clock::now();
-    let builder = FmtSubscriber::builder()
-        .with_timer(Uptime { start_time })
-        .with_env_filter(filter)
-        .with_filter_reloading()
-        .with_ansi(cfg!(test));
-    let handle = LevelHandle {
-        inner: builder.reload_handle(),
-    };
-    let dispatch = Dispatch::new(builder.finish());
-
-    (dispatch, handle)
+
+    match format {
+        "JSON" => {
```

Comment:
I think we probably want these to be case-insensitive — `"json"` should probably not get the plain format.

Reply:
Thanks I did something that feels hacky `match format.as_ref().to_uppercase().as_ref() {` ... is there a prettier way?

Reply:
https://github.com/linkerd/linkerd2-proxy/pull/500/commits/59f85e968e72de4ac11725419143e4072f781554